Custom ConfigSourceInterceptor makes build fail

### Describe the bug

Adding a custom ConfigSourceInterceptor to a quarkus app makes the build fail with the following stack trace during the `generate-code` phase:
```
Caused by: java.util.ServiceConfigurationError: io.smallrye.config.ConfigSourceInterceptor: Provider com.mycompany.config.MyConfigSourceInterceptor not found
    at java.util.ServiceLoader.fail (ServiceLoader.java:593)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass (ServiceLoader.java:1219)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService (ServiceLoader.java:1228)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext (ServiceLoader.java:1273)
    at java.util.ServiceLoader$2.hasNext (ServiceLoader.java:1309)
    at java.util.ServiceLoader$3.hasNext (ServiceLoader.java:1393)
    at io.smallrye.config.SmallRyeConfigBuilder.discoverInterceptors (SmallRyeConfigBuilder.java:131)
    at io.smallrye.config.SmallRyeConfig$ConfigSources.buildInterceptors (SmallRyeConfig.java:568)
    at io.smallrye.config.SmallRyeConfig$ConfigSources.<init> (SmallRyeConfig.java:516)
    at io.smallrye.config.SmallRyeConfig.<init> (SmallRyeConfig.java:66)
    at io.smallrye.config.SmallRyeConfigBuilder.build (SmallRyeConfigBuilder.java:419)
    at io.quarkus.deployment.CodeGenerator.lambda$trigger$1 (CodeGenerator.java:98)
    at io.quarkus.deployment.CodeGenerator.callWithClassloader (CodeGenerator.java:70)
    at io.quarkus.deployment.CodeGenerator.trigger (CodeGenerator.java:91)
    at io.quarkus.deployment.CodeGenerator.initAndRun (CodeGenerator.java:34)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at io.quarkus.maven.GenerateCodeMojo.generateCode (GenerateCodeMojo.java:79)
    at io.quarkus.maven.GenerateCodeMojo.doExecute (GenerateCodeMojo.java:54)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:123)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)

```

This is expected since at this point the class defined in the service loader file has not been compiled yet as discussed with @radcortez at https://quarkusio.zulipchat.com/#narrow/stream/294206-smallrye/topic/configsourceinterceptor

### Expected behavior

The quarkus maven plugin does not try to load custom ConfigSourceProvider during application build.

### Actual behavior

The quarkus maven plugin tries to load the custom ConfigSourceProvider that are supposed to be compiled later.

### How to Reproduce?

1. Create a project
2. Add a custom ConfigSourceProvider
3. Run `mvn clean package`

I tried reproducing it in a dedicated test project but without success, I have not been able to identify which conditions would reproduce the issue.

### Output of `uname -a` or `ver`

Darwin XXXX 21.2.0 Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "17.0.1" 2021-10-19 OpenJDK Runtime Environment GraalVM CE 21.3.0 (build 17.0.1+12-jvmci-21.3-b05) OpenJDK 64-Bit Server VM GraalVM CE 21.3.0 (build 17.0.1+12-jvmci-21.3-b05, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -43,7 +43,7 @@
         <microprofile-jwt.version>1.2</microprofile-jwt.version>
         <microprofile-lra.version>1.0</microprofile-lra.version>
         <smallrye-common.version>1.8.0</smallrye-common.version>
-        <smallrye-config.version>2.8.1</smallrye-config.version>
+        <smallrye-config.version>2.8.2</smallrye-config.version>
         <smallrye-health.version>3.1.2</smallrye-health.version>
         <smallrye-metrics.version>3.0.4</smallrye-metrics.version>
         <smallrye-open-api.version>2.1.17</smallrye-open-api.version>

--- a/core/deployment/src/main/java/io/quarkus/deployment/CodeGenerator.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/CodeGenerator.java
@@ -92,7 +92,11 @@ public class CodeGenerator {
 
             final PropertiesConfigSource pcs = new PropertiesConfigSource(properties, "Build system");
 
+            // Discovered Config classes may cause issues here, because this goal runs before compile
             final SmallRyeConfig config = ConfigUtils.configBuilder(false, false, launchMode)
+                    .setAddDiscoveredSources(false)
+                    .setAddDiscoveredInterceptors(false)
+                    .setAddDiscoveredConverters(false)
                     .withProfile(launchMode.getDefaultProfile())
                     .withSources(pcs)
                     .build();

